addons/udes_stock/models/stock_picking.py

```diff
@@ -532,6 +537,9 @@ def create_picking(
             # picking.move_line_ids.mapped('result_package_id').write({'package_id': False})
             pass
 
+        if create_batch:
+            batch = PickingBatch.create_batch(picking_type_id,None)
```

Comment:
This should explicitly create a batch with only the picking that was just created. I haven't looked at the implementation of create_batch, but I think it can potentially pick the oldest unbatched picking of the given type. If there are other Internal Transfers in the system outside batches then the user might get a different one than the one that was just created.